src/tasks/cosplay/array/list/ArrayNotList.java

```diff
@@ -0,0 +1,127 @@
+package tasks.cosplay.array.list;
+
+import java.util.Arrays;
+
+public class ArrayNotList<T> implements NotList<T> {
+
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] array;
+    private int indexForNextPut = 0;
+
+    public ArrayNotList() {
+        array = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    public void arrayGrower() {
+        int tempLength = array.length + (array.length >> 1);
+        T[] temp = (T[]) new Object[tempLength];
+        System.arraycopy(array, 0, temp, 0, array.length);
+        array = temp;
+    }
+
+    @Override
+    public void add(T value) {
+        if (indexForNextPut == array.length) {
+            arrayGrower();
+        }
+        array[indexForNextPut] = value;
+        indexForNextPut++;
+    }
+
+    @Override
+    public void add(T value, int index) {
+        if (index > indexForNextPut) {
+            throw new IndexOutOfBoundsException("Index Out Of Bounds");
+        }
+        T[] temp = (T[]) new Object[indexForNextPut - index];
+        System.arraycopy(array, index, temp, 0, temp.length);
+        array[index] = value;
+        System.arraycopy(temp, 0, array, index + 1, temp.length);
+        indexForNextPut++;
+    }
+
+    @Override
+    public void addAll(NotList<T> notList) {
+        while (indexForNextPut + notList.size() >= array.length) {
+            arrayGrower();
+        }
+        indexForNextPut += notList.size();
+        for (int i = 0; i < notList.size(); i++) {
+            add(notList.get(i));
+        }
+    }
+
+    @Override
+    public T get(int index) {
+        if (index >= indexForNextPut) {
+            throw new IndexOutOfBoundsException("Index Out Of Bounds");
+        }
+        return array[index];
+    }
+
+    @Override
+    public void set(T value, int index) {
+        if (index >= indexForNextPut) {
+            throw new IndexOutOfBoundsException("Index Out Of Bounds");
+        }
+        array[index] = value;
+    }
+
+    @Override
+    public T remove(int index) {
+        if (index >= indexForNextPut) {
+            throw new IndexOutOfBoundsException("Index Out Of Bounds");
+        }
+        T value = null;
+        for (int i = index; i < indexForNextPut; i++) {
+            if (value == null) {
```

Comment:
better remove this if statement from here and write at the end: return array[index];